### PR TITLE
OSDOCS-5702:updates ovn-k docs to include egressIP config object

### PR DESCRIPTION
**Pull request by JoeAldinger**
[OSDOCS-5702](https://issues.redhat.com//browse/OSDOCS-5702):updates ovn-k docs to include `egressIPConfig` object

Version(s):
4.12+

Issue:
https://issues.redhat.com/browse/OSDOCS-5702

Link to docs preview:
https://60822--docspreview.netlify.app/openshift-enterprise/latest/networking/ovn_kubernetes_network_provider/configuring-egress-ips-ovn.html#nw-egress-ips-config-object_configuring-egress-ips-ovn

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/nw-egress-ips-config-object.adoc [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+[id="nw-egress-ips-config-object_{context}"]
+== EgressIPconfig object
+As a feature of egress IP, the `reachabilityTotalTimeoutSeconds` configures the total timeout for checks that are sent by probes to egress IP nodes. The`egressIPConfig` object allows users to set the `reachabilityTotalTimeoutSeconds` `spec`. If the EgressIP node cannot be reached within this timeout, the node is declared down.
+
+You can increase this value if your network is not stable enough to handle the current default value of 1 second.
+
+The following YAML describes changing the `reachabilityTotalTimeoutSeconds` from the default 1 second probes to 5 second probes:
+
+[source,yaml]
```
Comment (skrthomas) [inline modules/nw-egress-ips-config-object.adoc:13]: I wonder if you may need to add a bit more context to this YAML, for example: 
```suggestion
[source,yaml]
apiVersion: k8s.ovn.org/v1
kind: EgressIP
  name: networks.operator.openshift.io
spec:
```

Comment (skrthomas) [reply modules/nw-egress-ips-config-object.adoc:13]: I'm not 100% on these suggested values, but typically YAML specs have some context like apiversion, kind, name before the spec.

File: modules/nw-egress-ips-config-object.adoc [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    hostPrefix: 23
+  defaultNetwork:
+    ovnKubernetesConfig:
+      egressIPConfig: <1>
+        reachabilityTotalTimeoutSeconds: 5 <2>
+      gatewayConfig:
+        routingViaHost: false
+      genevePort: 6081
+----
+<1> `egressIPConfig` holds the configurations for the options of the `EgressIP` object. Changing these configurations allows you to extend the `EgressIP` network object.
```
Comment (skrthomas) [inline modules/nw-egress-ips-config-object.adoc:27]:
```suggestion
<1> The `egressIPConfig` holds the configurations for the options of the `EgressIP` object. Changing these configurations allows you to extend the `EgressIP` network object.
```

File: modules/nw-egress-ips-config-object.adoc [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+<2> The value for `reachabilityTotalTimeoutSeconds` accepts integers from 0 to 60. A value of 0 disables the egressIP node's reachability check. Values of 1 to 60 correspond to the duration in seconds between probes sending the node's reachability check.
```
Comment (skrthomas) [inline modules/nw-egress-ips-config-object.adoc:29]:
```suggestion
<2> The value for `reachabilityTotalTimeoutSeconds` accepts integer values from 0 to 60. A value of 0 disables the egressIP node's reachability check. Values of 1 to 60 correspond to the duration in seconds between probes sending the node's reachability check.
```

Comment (skrthomas) [inline modules/nw-egress-ips-config-object.adoc:29]:
```suggestion
<2> The value for `reachabilityTotalTimeoutSeconds` accepts integers from 0 to 60. A value of 0 disables the reachability check of the egressIP node. Values of 1 to 60 correspond to the duration in seconds between probes sending the reachability check  for the node.
```

Comment (skrthomas) [reply modules/nw-egress-ips-config-object.adoc:29]: I think that instead of having node be possessive, you can rearrange the sentence a little.

**Review by skrthomas — COMMENTED** [review]
A few comments for your consideration :) nice work!

File: modules/nw-egress-ips-config-object.adoc [hunk 4]
```diff
@@ -0,0 +1,34 @@
+// Module included in the following assemblies:
+//
+// * networking/ovn_kubernetes_network_provider/assigning-egress-ips-ovn.adoc
+
+[id="nw-egress-ips-config-object_{context}"]
+= EgressIPconfig object
+As a feature of egress IP, the `reachabilityTotalTimeoutSeconds` configures the total timeout for checks that are sent by probes to egress IP nodes. The`egressIPConfig` object allows users to set the `reachabilityTotalTimeoutSeconds` `spec`. If the EgressIP node cannot be reached within this timeout, the node is declared down.
```
Comment (mburke5678) [inline modules/nw-egress-ips-config-object.adoc:7]: s/`reachabilityTotalTimeoutSeconds`/`reachabilityTotalTimeoutSeconds` parameter (or whatever)

File: modules/nw-egress-ips-config-object.adoc [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    hostPrefix: 23
+  defaultNetwork:
+    ovnKubernetesConfig:
+      egressIPConfig: <1>
+        reachabilityTotalTimeoutSeconds: 5 <2>
+      gatewayConfig:
+        routingViaHost: false
+      genevePort: 6081
+----
+<1> The `egressIPConfig` holds the configurations for the options of the `EgressIP` object. Changing these configurations allows you to extend the `EgressIP` network object.
```
Comment (mburke5678) [inline modules/nw-egress-ips-config-object.adoc:30]: Are the _the `EgressIP` object_ and _the `EgressIP` network object_ the same thing?

File: modules/nw-egress-ips-config-object.adoc (hunk 4 above)
Comment (mburke5678) [reply modules/nw-egress-ips-config-object.adoc:7]: s/The`egressIPConfig`/The `egressIPConfig`

Add a space; the marked up text isn't rendering correctly.

File: modules/nw-egress-ips-config-object.adoc [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+<2> The value for `reachabilityTotalTimeoutSeconds` accepts integer values from 0 to 60. A value of 0 disables the reachability check of the egressIP node. Values of 1 to 60 correspond to the duration in seconds between probes sending the reachability check for the node.
```
Comment (mburke5678) [inline modules/nw-egress-ips-config-object.adoc:32]: It seems that when we state things like _A value of_, the example values take markup, more often than not. 

> ...values from `0` to `60`. A value of `0` disables the reachability check of the egressIP node. Values of `1` to `60` correspond to the duration in seconds between probes sending the reachability check for the node

**Comment by mburke5678** [comment]
@JoeAldinger Sorry. A few peer-review-level things. When/if you fix them up, let me know and we can merge ASAP!

**Comment by huiran0826** [comment]
@jechen0648  Would you take a look ? It is about 'reachabilityTotalTimeoutSeconds'

File: modules/nw-egress-ips-config-object.adoc [hunk 7]
```diff
@@ -0,0 +1,34 @@
+// Module included in the following assemblies:
+//
+// * networking/ovn_kubernetes_network_provider/assigning-egress-ips-ovn.adoc
+
+[id="nw-egress-ips-config-object_{context}"]
+= EgressIPconfig object
```
Comment (ousleyp) [inline modules/nw-egress-ips-config-object.adoc:6]: Small nit, but the capitalization isn't consistent between the title and the body text. Should this be `egressIPConfig object`?

File: modules/nw-egress-ips-config-object.adoc [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    hostPrefix: 23
+  defaultNetwork:
+    ovnKubernetesConfig:
+      egressIPConfig: <1>
+        reachabilityTotalTimeoutSeconds: 5 <2>
+      gatewayConfig:
+        routingViaHost: false
+      genevePort: 6081
+----
+<1> The `egressIPConfig` holds the configurations for the options of the `EgressIP` object. Changing these configurations allows you to extend the `EgressIP` object.
```
Comment (ousleyp) [inline modules/nw-egress-ips-config-object.adoc:30]: I would add "stanza" or "field" after `egressIPConfig` here

**Review by ousleyp — COMMENTED** [review]
Overall, this looks good, so I'm going to merge it. Just a few suggestions for future improvements :) Thanks!

**Pull request merged**